akka-actor/src/main/scala/akka/io/dns/internal/DnsClient.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.io.dns.internal
+
+import java.net.{ InetAddress, InetSocketAddress }
+
+import akka.actor.Status.Failure
+import akka.actor.{ Actor, ActorLogging, ActorRef, NoSerializationVerificationNeeded, Stash }
+import akka.annotation.InternalApi
+import akka.io.dns.{ DnsRecordClass, DnsRecordType, DnsResourceRecord }
+import akka.io.{ IO, Udp }
+
+import scala.collection.immutable
+import scala.util.Try
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object DnsClient {
+  sealed trait Question {
+    def id: Short
+  }
+  final case class SrvQuestion(id: Short, name: String) extends Question
+  final case class Question4(id: Short, name: String) extends Question
+  final case class Question6(id: Short, name: String) extends Question
+  final case class Answer(id: Short, rrs: immutable.Seq[DnsResourceRecord]) extends NoSerializationVerificationNeeded
+  final case class DropRequest(id: Short)
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] class DnsClient(ns: InetSocketAddress) extends Actor with ActorLogging with Stash {
+
+  import DnsClient._
+
+  import context.system
+
+  IO(Udp) ! Udp.Bind(self, new InetSocketAddress(InetAddress.getByAddress(Array.ofDim(4)), 0))
+
+  var requests: Map[Short, ActorRef] = Map.empty
```

Comment:
If request message is lost will we be leaking memory here since there is no response and maybe no command failed?

Reply:
`AsyncDnsResolver` times out and sends a DropRequest.